pkg/server/proto/server.proto

```diff
@@ -137,6 +137,10 @@ service Waypoint {
   // GetDeployment returns a deployment
   rpc GetDeployment(GetDeploymentRequest) returns (Deployment) {}
 
+  // GetLatestDeployment returns the most recent successfully completed build
+  // for an app.
+  rpc GetLatestDeployment(GetLatestDeploymentRequest) returns (Deployment) {}
```

Comment:
This should be a `GetLatestDeploymentResponse` (a new proto message). It will mean this endpoint will be much more flexible for any clients consuming this endpoint. Otherwise the moment the response format changes, all clients will break who assume they should be receiving a Deployment proto message.

Reply:
That makes sense.

However, every other `GetLatestXYZ` endpoint simply returns XYZ so I'm wondering if it's confusing for `GetLatestDeployment` to not return `Deployment`?

Reply:
Hm, if the other ones return the bare Deployment proto then I guess we should stick to convention.

The idea with a special response message means it's much more flexible to include other proto messages without breaking all of the consuming clients of the API. If it's only a Deployment proto message, if you ever wanted to add status reports proto message for example, you'd have to go through the entire project and fix the response format to add a new field that was expecting the response type to only be a Deployment proto.